todo_app/app.py

```diff
@@ -8,162 +8,215 @@
-		view_model = ViewModel(my_objects, doing_objects,done_objects)
-		view_model.show_all_done_items()
-		 
-
-		return render_template("index.html", view_model=view_model )
-	@app.route('/create', methods =['POST'])
-	def create():
- 		title = request.form.get('title')
-
- 		newToDoCard(title)
- 		return redirect(url_for('index'))
-
-	@app.route('/complete_item', methods =['PUT'])
-	def complete_item(id):
- 		app.logger.info('Processing default request')
- 		print(id)
- 		updateCardToDone(id)
-
- 		return redirect(url_for('index'))
-
-	@app.route('/update', methods =['POST'])
-	def update():
- 		app.logger.info('Processing default request')
- 		id = request.form.get('id')
- 		print(id)
- 		return complete_item(id)
-
-	@app.route('/return_item', methods =['PUT'])
-	def return_item(n):
- 		app.logger.info('Processing default request')
- 		print(n)
- 		returnCardToDo(n)
- 		return redirect(url_for('index'))
-
-	@app.route('/update_back', methods =['POST'])
-	def update_back():
-		app.logger.info('Processing default request')
-		n = request.form.get('n')
-		print(n)
-		return return_item(n)
-	
-	
-	if __name__ == '__main__':
-
-		app.run(debug=True, host='0.0.0.0',port=5001)
-		
-	return app
-
-
-
-
-
-
+    app = Flask(__name__)
+
+    toDoId = os.environ['TOID']
+    doingId = os.environ['DOINGID']
+    doneId = os.environ['DONE']
+    dbconnect = os.environ['CLIENT']
+    githubId = os.environ['GITHUBID']
+    webClient = os.environ['WEBAPPLICATIONCLIENT']
+    clientSecret = os.environ['CLIENTSECRET']
+    todoBoard = os.environ['TODOBOARD']
+    doingBoard = os.environ['DOINGBOARD']
+    doneBoard = os.environ['DONEBOARD']
+    app.secret_key = os.getenv('SECRET_KEY')
+
+    def connectDb():
+        client = pymongo.MongoClient(
+            dbconnect, ssl=True, ssl_cert_reqs='CERT_NONE')
+        # print(client.list_database_names())
+        database = client["01"]
+        trello_collection = database["trello_collection"]
+        todo = trello_collection.find(
+            {'idBoard': todoBoard}, {"name": 1})
+        for x in todo:
+            print(x)
+
+        doing = trello_collection.find(
+            {'idBoard': doingBoard}, {"name": 1})
+        for x in todo:
+            print(x)
+
+        done = trello_collection.find(
+            {'idBoard': doneBoard}, {"name": 1})
+        for x in doing:
+            print(x)
+
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        client = WebApplicationClient(webClient)
+        full_redirect_url = client.prepare_request_uri(
+            'https://github.com/login/oauth/authorize', redirect_uri='http://localhost:5000/callback')
+
+        return redirect(full_redirect_url)
+
+    @login_manager.user_loader
+    def load_user(id):
+
+        if id == githubId:
+            role = "writer"
+        else:
+            role = "reader"
+        id = User(id, role)
+        print(role)
+
+        return id
```

Comment:
It's confusing to re-use the `id` variable. To start with, `id` is a string (the user's id) and then by the end of the function, `id` is a full User object.

Reply:
Updated, thank you